vllm/v1/core/sched/scheduler.py

```diff
@@ -846,8 +848,14 @@ def finish_requests(
 
         for req_id in request_ids:
             request = self.requests.get(req_id)
+            # Request is already finished.
             if request is None:
-                # Invalid request ID.
+                # Abort pending free kv reqs depending on connector impl.
+                should_free = (self.connector is None or
+                               self.connector.should_free_pending_on_abort())
+                if (req_id in self.pending_kv_free_req_ids and should_free):
```

Comment:
A minor optimization
```suggestion
                if should_free and req_id in self.pending_kv_free_req_ids:
```